zipline/finance/performance/tracker.py

```diff
@@ -333,40 +329,37 @@ def handle_minute_close(self, dt, data_portal):
                                             account.leverage)
 
         minute_packet = self.to_dict(emission_type='minute')
+        return minute_packet
 
-        # if this is the close, update dividends for the next day.
-        # Return the performance tuple
-        if dt == self.market_close:
-            return minute_packet, self._handle_market_close(
-                todays_date, data_portal._adjustment_reader,
-            )
-        else:
-            return minute_packet, None
-
-    def handle_market_close_daily(self, dt, data_portal):
+    def handle_market_close(self, dt, data_portal):
         """
-        Function called after handle_data when running with daily emission
-        rate.
+        Handles the close of the given day, in both minute and daily emission.
+        In daily emission, also updates performance, benchmark and risk metrics
+        as it would in handle_minute_close if it were minute emission.
+
+        Parameters
+        __________
+        dt : Timestamp
+            The minute that is ending
+
+        Returns
+        _______
+        A daily perf packet.
         """
-        self.position_tracker.sync_last_sale_prices(dt, False, data_portal)
-        self.update_performance()
         completed_date = self.day
-        account = self.get_account(False)
-
-        benchmark_value = self.all_benchmark_returns[completed_date]
 
-        self.cumulative_risk_metrics.update(
-            completed_date,
-            self.todays_performance.returns,
-            benchmark_value,
-            account.leverage)
+        if self.emission_rate == 'daily':
+            self.position_tracker.sync_last_sale_prices(dt, False, data_portal)
```

Comment:
the method that was previously `handle_market_close_daily` and contains this block was only called in daily mode. I suspect running this block would not effectively update anything, as that would have been done in the latest `handle_minute_close`. Additionally, I believe lines like `benchmark_value = self.all_benchmark_returns[completed_date]` would crash if called in minute emission, since the `completed_date` is normalized and the benchmark would not have midnight